File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        res = float("inf")
        l = 0
        total = 0
        for r in range(n):
            total += nums[r]
            while total >= target:
                res = min(r- l + 1, res)
                total -= nums[l]
                l += 1

        return 0 if res == float("inf") else res

File: Data Structures & Algorithms/jump-game/submission-1.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int greedy = nums.size() - 1;
        for (int i = nums.size() -2; i >= 0; i--){
            if(nums[i] + i >= greedy){
                greedy = i;
            }
        }
        return greedy == 0;
    }
};
